Go staticcheck failing

```
internal/interpreter/parser/parser.go:116:3: QF1003: could use tagged switch on token.Kind (staticcheck)
		if token.Kind == "[" {
		^
internal/jsone.go:291:17: ST1023: should omit type string from declaration; it will be inferred from the right-hand side (staticcheck)
		var seperator string = ""
		              ^
2 issues:
* staticcheck: 2
```

(also, it's spelled "separator"!)

--- a/internal/interpreter/parser/parser.go
+++ b/internal/interpreter/parser/parser.go
@@ -113,12 +113,13 @@ func (p *Parser) parsePropertyAccessOrFunc() (node IASTNode, err error) {
 	operators := []string{"[", "(", "."}
 
 	for token := p.CurrentToken; p.CurrentToken != (Token{}) && StringsContains(token.Kind, operators); token = p.CurrentToken {
-		if token.Kind == "[" {
+		switch token.Kind {
+		case "[":
 			node, err = p.parseAccessWithBrackets(node)
 			if err != nil {
 				return nil, err
 			}
-		} else if token.Kind == "." {
+		case ".":
 			token = p.CurrentToken
 			err = p.takeToken(".")
 			if err != nil {
@@ -132,7 +133,7 @@ func (p *Parser) parsePropertyAccessOrFunc() (node IASTNode, err error) {
 			}
 			binaryNode.NewNode(token, node, rightPart)
 			node = binaryNode
-		} else if token.Kind == "(" {
+		case "(":
 			node, err = p.parseFunctionCall(node)
 			if err != nil {
 				return nil, err

--- a/internal/jsone.go
+++ b/internal/jsone.go
@@ -273,7 +273,7 @@ var builtin = map[string]interface{}{
 	"join": i.WrapFunction(func(v []interface{}, sep ...interface{}) (string, error) {
 		// We use variadic because golang doesn't support optional parameters
 		if len(sep) > 1 {
-			return "", fmt.Errorf("join(value, seperator) takes at-most two arguments, but was given %d", len(sep))
+			return "", fmt.Errorf("join(value, separator) takes at-most two arguments, but was given %d", len(sep))
 		}
 
 		var reference interface{} = nil
@@ -284,14 +284,14 @@ var builtin = map[string]interface{}{
 			_, isString := reference.(string)
 
 			if !isString && !isNumber {
-				return "", fmt.Errorf("join(value, seperator) seperator must be a string or a number")
+				return "", fmt.Errorf("join(value, separator) separator must be a string or a number")
 			}
 		}
 
-		var seperator string = ""
+		separator := ""
 
 		if reference != nil {
-			seperator = fmt.Sprintf("%v", reference)
+			separator = fmt.Sprintf("%v", reference)
 		}
 
 		valuesText := []string{}
@@ -307,12 +307,12 @@ var builtin = map[string]interface{}{
 			case float64:
 				text = fmt.Sprintf("%v", item)
 			default:
-				return "", fmt.Errorf("join(value, seperator) value must be a list of strings or numbers")
+				return "", fmt.Errorf("join(value, separator) value must be a list of strings or numbers")
 			}
 			valuesText = append(valuesText, text)
 		}
 
-		return strings.Join(valuesText, seperator), nil
+		return strings.Join(valuesText, separator), nil
 	}),
 	"fromNow": i.WrapFunctionWithContext(func(context map[string]interface{}, offset string, from ...string) (string, error) {
 		// We use variadic because golang doesn't support optional parameters
@@ -705,7 +705,7 @@ var operators = map[string]operator{
 			additionalContextVars = 3
 		}
 		eachTemplate := template[eachKey]
-		
+
 		switch val := value.(type) {
 		case []interface{}:
 			var result interface{}

